hydra-node/src/Hydra/Ledger.hs

```diff
@@ -25,27 +25,21 @@ type Party = Natural
 class
   ( Eq tx
   , Eq (UTxO tx)
-  , Eq (LedgerState tx)
   , Show tx
   , Show (UTxO tx)
-  , Show (LedgerState tx)
   ) =>
   Tx tx
   where
   type UTxO tx
-  type LedgerState tx
 
 data Ledger tx = Ledger
-  { canApply :: LedgerState tx -> tx -> ValidationResult
-  , applyTransaction :: LedgerState tx -> tx -> Either ValidationError (LedgerState tx)
-  , initLedgerState :: LedgerState tx
-  , fromUTxO :: UTxO tx -> LedgerState tx
-  , getUTxO :: LedgerState tx -> UTxO tx
+  { applyTransactions :: UTxO tx -> [tx] -> Either ValidationError (UTxO tx)
+  , initUTxO :: UTxO tx
```

Comment:
It feels a bit weird to have `initUTxO` in here now. Following a baby-step approach it's fine but I think we should move it out of this interface moving forward, and move it under the `Tx` type-class.